File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 3/4_maximum_advertisement_revenue/dot_product.py
def ad_revenue(num, profit, clicks):
    profit.sort()
    clicks.sort()
    revenue = 0
    for i in range(num):
        revenue += profit[i] * clicks[i]
    return revenue


if __name__ == '__main__':
    num = int(input())
    profit = [int(x) for x in input().split()]
    clicks = [int(x) for x in input().split()]
    print(ad_revenue(num, profit, clicks))

File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 3/4_maximum_advertisement_revenue/dot_product.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int ad_revenue(int num, vector<int> &profit, vector<int> &clicks)
{
    int revenue = 0;
    sort(profit.begin(), profit.end());
    sort(clicks.begin(), clicks.end());
    for (int i = 0; i < num; i++)
    {
        revenue += profit[i] * clicks[i];
    }
    return revenue;
}

int main()
{
    int num, temp;
    cin >> num;
    vector<int> profit;
    vector<int> clicks;
    for (int i = 0; i < num; i++)
    {
        cin >> temp;
        profit.push_back(temp);
    }
    for (int i = 0; i < num; i++)
    {
        cin >> temp;
        clicks.push_back(temp);
    }
    cout << ad_revenue(num, profit, clicks) << "\n";
    return 0;
}
